packages/optimizers/terser/src/TerserOptimizer.js

```diff
@@ -36,45 +37,24 @@ export default new Optimizer({
       },
       sourceMap: {
         filename: path.relative(options.projectRoot, bundle.filePath),
+        asObject: true,
+        content: originalMap,
       },
       module: bundle.env.outputFormat === 'esmodule',
     };
 
-    let sourceMap = null;
-    if (options.sourceMaps) {
-      sourceMap = new SourceMap();
-      // $FlowFixMe
-      config.output = {
-        source_map: {
-          add(source, gen_line, gen_col, orig_line, orig_col, name) {
-            // $FlowFixMe
-            sourceMap.addMapping({
-              source,
-              name,
-              original: {
-                line: orig_line,
-                column: orig_col,
-              },
-              generated: {
-                line: gen_line,
-                column: gen_col,
-              },
-            });
-          },
-        },
-      };
-    }
-
-    if (sourceMap && map) {
-      sourceMap = await map.extend(sourceMap);
-    }
-
+    // $FlowFixMe, config has outdated sourcemap typings
```

Comment:
Could you update them? We are using our own typings anyway
https://github.com/parcel-bundler/parcel/blob/v2/flow-libs/terser.js.flow

Reply:
ow yeah sure